### ISSUE TITLE
Drone missing nsswitch.conf

### ISSUE THREAD

**Issue by matt-simons** [drone/drone#2793]
When pulling drone:1.1.0 there seems to be a regression from when this was last fixed in #1306

Drone no longer seems to respect /etc/hosts.

**Comment by bradrydzewski** [comment]
I cannot remember the exact reasons, but the the nsswitch was removed back in 2016

**Comment by tboerger** [comment]
Drone is based on alpine, if I remember correctly Müsli libc didn't rely on nsswitch.conf

**Comment by matt-simons** [comment]
If I understand [this](https://giedrius.blog/2018/11/22/etc-nsswitch-conf-and-etc-hosts-woes-with-the-alpine-and-others-docker-image-and-golang/) correctly, it is golang's net package which assumes there will always be an nsswitch, which isn't always the case.

Which I suppose leaves 3 options:

1. Add an nsswitch anyway so that drone will read /etc/hosts (though as @bradrydzewski suggests this may cause other issues)
2. Bake support for reading /etc/hosts into drone itself
3. Status quo and require the end user to mount an nsswitch.conf file at runtime

It would be ideal if the drone images supported Docker linking and Kubernetes hostAlias out of the box.

**Comment by jstxu** [comment]
I can confirm that running drone with the flag `-v /etc/nsswitch.conf:/etc/nsswitch.conf` makes it work. 

So I've made a PR: https://github.com/drone/drone/pull/2820

**Comment by jstxu** [comment]
I wanted to add the `nsswitch.conf` because I needed to setup the gitea service ip on the `/etc/hosts`. 

If this is your case, just using the steps above won't be enough, as even though the Drone server will access the correct IP, the pipeline slaves won't because those dockers won't have the `nsswitch.conf` either.

So the solution I've found is to use the host dns server for the containers. You can
specify the dns servers in the docker networks, but if you don't want to specify the `--network` flag on the drone slaves, you have to modify the `bridge0` network. To do so you have to add this configuration to the `/etc/docker/daemon.json`

```json
{
  "dns": ["172.17.0.1"]
}
```

Usually systemd machines have a `systemd-resolv` service listening on
`127.0.0.53:53`, the problem is that you can't access that address from within
the container, so you would have to make systemd-resolv listen to the inet
address of the bridge0 network, or use another dns server that forwards the dns
queries from the docker network to the systemd one or the `/etc/hosts` file.

```bash
$ ifconfig docker0
docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
...
```

I've found easier to do the second one, so you just have to install dnsmasq,
configure the listenaddress in `/etc/dnsmasq.conf`.

```
listen-address=127.0.0.1,172.17.0.1
```
and restart both `dnsmasq` and `docker` services.

**Comment by bradrydzewski** [comment]
closed by f6ba3ea7b2aa10dfc79ca43c156ac2924ba1e039
